index.js

```diff
@@ -1,114 +1,138 @@
+   */
+  function render (indexUrl, output) {
+    var png = output.toLowerCase().endsWith('.png')
+
+    var pdfOptions = {
+      marginsType: argv.marginsType,
+      printBackground: argv.printBackground,
+      printSelectionOnly: argv.printSelectionOnly,
+      pageSize: argv.pageSize,
+      landscape: argv.landscape
+    }
 
-    if (customCss) {
-      opts.customCss = customCss
+    var pageDimensions = {
+      'A4': {x: 595, y: 842},
+      'Letter': {x: 8.5 * HTML_DPI, y: 11 * HTML_DPI},
+      'Legal': {x: 8.5 * HTML_DPI, y: 14 * HTML_DPI}
     }
 
-    // if given a markdown, render it into HTML and return the path of the HTML
-    input = markdownToHTMLPath(input, opts, function (err, tmpHTMLPath) {
-      if (err) {
-        console.error('Parse markdown file error', err)
-        app.quit()
+    var pageDim = pageDimensions[pdfOptions.pageSize]
+    pageDim = pdfOptions.landscape ? {x: pageDim.y, y: pageDim.x} : pageDim
+    var winX = argv.windowX || pageDim.x
+    var winY = argv.windowY || pageDim.y
+
+    console.info('Opening a browser window', winX, 'x', winY)
+    var win = new BrowserWindow({
+      width: winX,
+      height: winY,
+      x: 0, y: 0,
+      show: true,
+      enableLargerThanScreen: true,
+      webPreferences: {
+        experimentalFeatures: false,
+        experimentalCanvasFeatures: false
       }
-
-      var indexUrl = wargs.urlWithArgs(tmpHTMLPath, {})
-      render(indexUrl, output)
     })
-  } else {
-    var indexUrl = wargs.urlWithArgs(input, {})
-    render(indexUrl, output)
-  }
-}
 
-/**
- * render file to pdf
- * @param  {String} indexUrl The path to the HTML or url
- */
-function render (indexUrl, output) {
-  var wait = argv.w || argv.outputWait || 0
-  var win = new BrowserWindow({ width: 0, height: 0, show: false })
-  win.on('closed', function () { win = null })
-
-  var loadOpts = {}
-  if (argv.d || argv.disableCache) {
-    loadOpts.extraHeaders = 'pragma: no-cache\n'
-  }
-  win.loadURL(indexUrl, loadOpts)
-
-  // print to pdf args
-  var opts = {
-    marginsType: argv.m || argv.marginsType || argv.marginType || 0,
-    printBackground: argv.b || argv.printBackground || true,
-    printSelectionOnly: argv.s || argv.printSelectionOnly || false,
-    pageSize: argv.p || argv.pageSize || 'A4',
-    landscape: argv.l || argv.landscape || false
-  }
+    win.on('closed', function () {
+      win = null
+    })
 
-  win.webContents.on('did-finish-load', function () {
-    setTimeout(function () {
-      win.webContents.printToPDF(opts, function (err, data) {
-        if (err) {
-          console.error(err)
+    var loadOpts = {}
+    if (argv.disableCache) {
+      loadOpts.extraHeaders = 'pragma: no-cache\n'
+    }
+    win.loadURL(indexUrl, loadOpts)
+
+    win.webContents.on('did-finish-load', function () {
+      setTimeout(function () {
+        console.log('finished loading')
+
+        //  Image (PNG)
+        if (png) {
```

Comment:
I need PNGS, figured others might too.  Activated if the output file ends with .png